nemo/core/classes/modelPT.py

```diff
@@ -1877,24 +1936,36 @@ def on_train_batch_end(self, outputs, batch: Any, batch_idx: int, unused: int =
         https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#on-train-batch-end
         We use it here to enable nsys profiling.
         """
-
+        # logging.info(f"Training Real batch {self._real_batch_idx} ended.")
         if self.device.type == 'cuda':
             if hasattr(self, '_nsys_profile_enabled'):
                 if self._nsys_profile_enabled and not self._nsys_profile_complete:
-                    if batch_idx >= self._nsys_profile_end_step and get_rank() in self._nsys_profile_ranks:
+                    if self._real_batch_idx >= self._nsys_profile_end_step and get_rank() in self._nsys_profile_ranks:
                         logging.info("====== End nsys profiling ======")
                         torch.cuda.cudart().cudaProfilerStop()
                         self._nsys_profile_complete = True
 
             if hasattr(self, '_memory_profile_enabled'):
                 if self._memory_profile_enabled and not self._memory_profile_complete:
-                    if batch_idx >= self._memory_profile_end_step and get_rank() == self._memory_profile_rank:
-                        logging.info("====== End CUDA memory profiling ======")
+                    if self._real_batch_idx == self._memory_profile_end_step and get_rank() == self._memory_profile_rank:
+                        logging.info(f"====== Rank[{self._memory_profile_rank}], Batch[{batch_idx}]. End CUDA memory profiling: Activation. Snapshot saved in {self._memory_profile_snapshot_file_activation} ======")
                         torch.cuda.memory._dump_snapshot(
-                            f'{self._memory_profile_output_path}/memory_profile_rank{self._memory_profile_rank}.pickle'
+                            self._memory_profile_snapshot_file_activation
                         )
                         torch.cuda.memory._record_memory_history(enabled=None)
                         self._memory_profile_complete = True
+                    # Call the analysis function
+                    if self._memory_profile_analysis_enabled and self._memory_profile_complete:
```

Comment:
same as previously, `self._memory_profile_analysis_enabled` should be true? due to line 1890